### PR TITLE
Oraclelinux docker build

### PR DESCRIPTION
**Pull request by jraman567**
Add support to build Veraison with Oracle Linux container

### CODE REVIEW

**Comment by jraman567** [comment]
I can see some CI failures; I'm taking a look at it.

**Review by setrofim — CHANGES_REQUESTED** [review]
Could you update the first commit message to elaborate on 

> But in some cases, Ubuntu docker containers aren't available.

The Ubuntu images are part of the public Docker registry, and should always be available. Adding a parallel set of dockerfiles would introduce a significant maintenance burden going forward, and I'd rather avoid that if at all possible.

----

When attempting to build the "ol" configuration locally, I get

```
[+] Building 1.3s (12/25)                                                                                                                                                                                                                                                                                       docker:default
 => [internal] load build definition from builder.docker                                                                                                                                                                                                                                                                  0.0s
 => => transferring dockerfile: 2.40kB                                                                                                                                                                                                                                                                                    0.0s
 => WARN: FromAsCasing: 'as' and 'FROM' keywords' casing do not match (line 1)                                                                                                                                                                                                                                            0.0s
 => [internal] load metadata for container-registry.oracle.com/os/oraclelinux:9                                                                                                                                                                                                                                           1.1s
 => [internal] load .dockerignore                                                                                                                                                                                                                                                                                         0.0s
 => => transferring context: 2B                                                                                                                                                                                                                                                                                           0.0s
 => [internal] load build context                                                                                                                                                                                                                                                                                         0.0s
 => => transferring context: 285.21kB                                                                                                                                                                                                                                                                                     0.0s
 => [ 1/21] FROM container-registry.oracle.com/os/oraclelinux:9@sha256:a6296b68f10b71197d1bb273424444e31f45525bc2531ddcfd23bd82fe17dfe3                                                                                                                                                                                   0.0s
 => CACHED [ 2/21] RUN cat /etc/redhat-release                                                                                                                                                                                                                                                                            0.0s
 => CACHED [ 3/21] RUN dnf config-manager --set-enabled ol9_codeready_builder                                                                                                                                                                                                                                             0.0s
 => CACHED [ 4/21] RUN dnf install -y git                                                                                                                                                                                                                                                                                 0.0s
 => CACHED [ 5/21] RUN dnf install -y make                                                                                                                                                                                                                                                                                0.0s
 => CACHED [ 6/21] RUN dnf update -y     && dnf install -y         protobuf*         sqlite  uuid  gettext         vim         jq         iputils         nmap         wget         sudo         ca-certificates         gcc     && uuidgen | tr -d - > /etc/machine-id     && rm -rf /var/tmp/* /tmp/*                   0.0s
 => CACHED [ 7/21] RUN groupadd -g 1001 builder &&     groupadd -g 616 veraison &&     useradd -m -u 1000 -g builder -G root,veraison,wheel -p paxyJoB0NZsNU -s /bin/bash builder                                                                                                                                         0.0s
 => ERROR [ 8/21] RUN groupadd -g 1001 opc                                                                                                                                                                                                                                                                                0.2s
------                                                                                                                                                                                                                                                                                                                         
 > [ 8/21] RUN groupadd -g 1001 opc:
0.150 groupadd: GID '1001' already exists
------

 1 warning found (use docker --debug to expand):
 - FromAsCasing: 'as' and 'FROM' keywords' casing do not match (line 1)
builder.docker:39
--------------------
  37 |         useradd -m -u ${BUILDER_UID} -g builder -G root,veraison,wheel -p paxyJoB0NZsNU -s /bin/bash builder
  38 |     
  39 | >>> RUN groupadd -g 1001 opc
  40 |     RUN useradd -m -u 1001 -g opc -G root,veraison,wheel -p paxyJoB0NZsNU -s /bin/bash opc
  41 |     
--------------------
ERROR: failed to solve: process "/bin/bash -c groupadd -g 1001 opc" did not complete successfully: exit code: 4
```

----

Regarding the logs volume subdirectory -- is this something specific to oraclelinux, or to a particular Docker version? This does not seem to be necessary on `main`.

If the subdirectory is unavoidable, please rename to something less confusing, e.g. "logs/services/", rather than "logs/logs/".

----

Additionally, please:

- move the distro-specific subdirs under src/ (use `SRC_DIR` rather than `THIS_DIR` as base)
- rename "ol" directory to something more obvious, e.g. "oraclelinux"; or alternatively, group it and "ubuntu" under an additional sub-directory (e.g. "dockerfiles" or "distros").
- squash the fix-up commits (if a commit only changes things introduced in a previous commit in the same pull, it should not be a separate commit).
- fix the FromAsCase warnings (capitalize "AS" in the dockerfiles)

File: deployments/docker/Makefile [hunk 1]
```diff
@@ -34,21 +34,31 @@ ifneq ($(DEBUG_PORT),)
 DEBUG_PORT_FLAG := -p $(DEBUG_PORT):$(DEBUG_PORT)
 endif
 
+# Determine OS distribution
+DISTRO := $(shell cat /etc/os-release | grep -w ID | cut -d '=' -f2 | cut -d '"' -f2)
```
[the remark this answers is not in the document]
Comment (setrofim) [reply deployments/docker/Makefile:38]: > I'll try to reproduce this. I assume your host is running Ubuntu. @setrofim Is that correct?
  
Arch 
Docker version 27.3.1, build ce1223035a

[the remark this answers is not in the document]
Comment (jraman567) [reply deployments/docker/Makefile:38]: Thanks

**Comment by jraman567** [comment]
@setrofim  and I discussed this. We concluded that docker isn't the official build process. Maintaining two sets of dockerfiles, one for docker and another for podman, is cumbersome. For these reasons, we decided to shelve this PR. Closing.

**Pull request closed without merging**